exist-core/src/main/java/org/exist/repo/Deployment.java

```diff
@@ -462,6 +478,8 @@ private void checkProcessorVersion(final PackageLoader.Version version) throws P
                 return Optional.ofNullable(targetCollection.getCollectionPath());
             }
         } catch (final XPathException e) {
+            // reset possibly altered trigger state
+            broker.setTriggersEnabled(triggerState);
```

Comment:
This isn't strong enough... consider using a try/finally mechanism

Reply:
You're right, thanks